lib/iris/tests/unit/analysis/test_COUNT.py

```diff
@@ -23,11 +23,51 @@
 # importing anything else.
 import iris.tests as tests
 
+import numpy as np
 import numpy.ma as ma
 
 from iris.analysis import COUNT
-import iris.cube
+from iris.cube import Cube
 from iris.coords import DimCoord
+from iris._lazy_data import as_lazy_data
+
+
+class Test_basics(tests.IrisTest):
+    def setUp(self):
+        data = np.array([1, 2, 3, 4, 5])
+        coord = DimCoord([6, 7, 8, 9, 10], long_name='foo')
+        self.cube = Cube(data)
+        self.cube.add_dim_coord(coord, 0)
+        self.lazy_cube = Cube(as_lazy_data(data))
+        self.lazy_cube.add_dim_coord(coord, 0)
+        self.func = lambda x: x >= 3
+
+    def test_name(self):
+        self.assertEqual(COUNT.name(), 'count')
+
+    def test_no_function(self):
+        with self.assertRaisesRegexp(KeyError, 'no selection function'):
+            self.lazy_cube.collapsed("foo", COUNT)
```

Comment:
I'm not that happy that the unit test is calling ``cube.collapsed``. Could you just test that the aggregator produces the right numbers when you pass a numpy array through, and thus rely on the testing of the Aggregator class to ensure that the integration is good?